src/main/java/command/PostponeCommand.java

```diff
@@ -32,53 +29,59 @@ public PostponeCommand(int indexOfTask, LocalDateTime startDate) {
     }
 
     /**
-     * Secondary constructor for event tasks.
+     * Secondary constructor for event tasks in particular.
      *
-     * @param indexOfTask Index of the task to postpone
-     * @param fromDate    New fromDate to represent the new time where event will start
-     * @param toDate      New toDate to represent the new time where the event will end
+     * @param indexOfTask Holds the index of the task to be commented on.
+     * @param fromDate    Holds the new start time of a task.
+     * @param toDate      Holds the new end time of a task.
      */
     public PostponeCommand(int indexOfTask, LocalDateTime fromDate, LocalDateTime toDate) {
         this.indexOfTask = indexOfTask;
         this.fromDate = fromDate;
         this.toDate = toDate;
     }
 
+    /**
+     * Postpones a task properly and saves the updated TaskList  it to persistent storage.
+     *
+     * @param tasks   Holds the list of all the tasks the user has.
+     * @param storage Allows the saving of the file to persistent storage.
+     */
     @Override
     public void execute(TaskList tasks, Storage storage) throws DukeException {
-        if (indexOfTask < 0 || indexOfTask > (tasks.getSize() - 1)) {
-            throw new DukeException(DukeException.taskDoesNotExist());
-        }
-        Task taskToBePostponed = tasks.getTasks().get(indexOfTask);
-        String description = taskToBePostponed.description;
+        if (isIndexValid(indexOfTask, tasks.getSize())) {
 
-        if (taskToBePostponed.toString().contains("[D]")) {
-            if (startDate == null) {
-                throw new DukeException(DukeException.wrongDateOrTime());
-            }
-            Deadline deadlineTest = new Deadline(description, startDate);
-            if (tasks.isClash(deadlineTest)) {
-                throw new DukeException(DukeException.taskClash());
-            } else {
-                taskToBePostponed.setStartDate(startDate);
-                storage.saveFile(tasks.getTasks());
-                Ui.printOutput("Got it! I've postponed this deadline:\n " + taskToBePostponed.toString());
-            }
-        } else if (taskToBePostponed.toString().contains("[E]")) {
-            if (fromDate == null || toDate == null) {
-                throw new DukeException(DukeException.wrongDateOrTime());
-            }
-            Event eventTest = new Event(description, fromDate, toDate);
-            if (tasks.isClash(eventTest)) {
-                throw new DukeException(DukeException.taskClash());
+            Task taskToBePostponed = tasks.getTasks().get(indexOfTask);
+            String description = taskToBePostponed.description;
+
+            if (taskToBePostponed.toString().contains("[D]")) {
```

Comment:
There needs to better way to check the task's class than this. In the event the task's toString function changes (ex. from [D] to [Deadline]), this class needs to be modified as well.

We'll refactor this eventually